### PR TITLE
rbdmap: systemd support

### PR DESCRIPTION
**Pull request by branto1**
Write an init-rbdmap init script. To achieve this, the map/ unmap functionality was moved to src/rbdmap and the template config was moved to src/etc-rbdmap.

### CODE REVIEW

**Comment by liewegas** [comment]
lgtm for rpm, we should add it to the debian package tho too

**Comment by liewegas** [comment]
...and retarget at master.  we'll backport to infernalis later, but it's too late for 9.2.0

**Comment by branto1** [comment]
Hm, it looks like the previous patch set that touched this did not make it into master but infernalis, only. These changes directly follow-up. I have opened another PR to fix this:

https://github.com/ceph/ceph/pull/6445

@liewegas: I tried to add some debian changes to accomodate that, I'll re-target to master once the other PR reaches master.

@tchaikov: Added Fixes: #13374 to the commit messages. Not sure about the bash dependency in debian control file but you don't need to specify a dependency on the packages that are in minimal install and bash is in that group in fedora so no need to change fedora .spec.

**Comment by tchaikov** [comment]
> Added Fixes: #13374 to the commit messages.

seems the updated commit was not pushed to your remote branch?

> Not sure about the bash dependency in debian control file

```patch
diff --git a/debian/control b/debian/control
index 4965952..a75c4d6 100644
--- a/debian/control
+++ b/debian/control
@@ -196,7 +196,8 @@ Depends: librbd1 (= ${binary:Version}), ${misc:Depends}, ${shlibs:Depends},
          python-rados (= ${binary:Version}),
          python-cephfs (= ${binary:Version}),
          python-rbd (= ${binary:Version}),
-        python-requests
+        python-requests,
+        bash (>= 2.04)
 Conflicts: ceph-client-tools
 Replaces: ceph-client-tools,
          ceph (<< 9.0.0-943),
```

please note that, i didn't test the bash (< 2.04) / check the release notes of bash. the reason why i put "(>= 2.0)", is just a safe bet.

**Comment by branto1** [comment]
@tchaikov: You're right. I was working on a different branch. It should be fixed, now. Also, I changed the bash shebang to the sh one. That should be ok as (afaik) debian runs init scripts with dash, not bash and it used to run the script just fine before.

File: src/Makefile-client.am [hunk 1]
```diff
@@ -48,7 +48,8 @@ bash_completion_DATA += \
 
 bin_SCRIPTS += \
 	ceph-rbdnamer \
-	rbd-replay-many
+	rbd-replay-many \
+        rbdmap
```
Comment (tchaikov) [inline src/Makefile-client.am]: @branto1 could you update src/CMakeLists.txt also?
```patch
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 28a02ed..05ff6af 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -884,6 +884,7 @@ if(${WITH_RBD})
     ${BLKID_LIBRARIES} ${CMAKE_DL_LIBS} ${TCMALLOC_LIBS})
   install(TARGETS rbd DESTINATION bin)
   install(PROGRAMS ${CMAKE_SOURCE_DIR}/src/ceph-rbdnamer DESTINATION bin)
+  install(PROGRAMS ${CMAKE_SOURCE_DIR}/src/rbdmap DESTINATION bin)

   set(librbd_replay_types_srcs
       rbd_replay/ActionTypes.cc)
```

would suffice, i guess.

**Comment by branto1** [comment]
Closing this PR to re-target to master.

**Comment by branto1** [comment]
The new PR targeted to master with the CMakeLists.txt change:

https://github.com/ceph/ceph/pull/6479

**Pull request closed without merging**